File: src/com/mishiranu/dashchan/content/net/firewall/FirewallUtils.java
package com.mishiranu.dashchan.content.net.firewall;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

final class FirewallUtils {

	private FirewallUtils() {

	}

	static Map<String, String> parseCookies(String cookiesString) {
		Map<String, String> cookies;
		if (cookiesString != null && !cookiesString.isEmpty()) {
			cookies = new HashMap<>();
			String[] splitted = cookiesString.split(";\\s*");
			for (String pair : splitted) {
				int index = pair.indexOf('=');
				if (index >= 0) {
					String key = pair.substring(0, index);
					String value = pair.substring(index + 1);
					cookies.put(key, value);
				}
			}
		} else {
			cookies = Collections.emptyMap();
		}
		return cookies;
	}

}

File: src/com/mishiranu/dashchan/content/net/firewall/FirewallUtils.kt
package com.mishiranu.dashchan.content.net.firewall

internal object FirewallUtils {
    @JvmStatic
    fun parseCookies(cookiesString: String?): Map<String, String> {
        if (cookiesString.isNullOrEmpty()) {
            return emptyMap()
        }
        val cookies = HashMap<String, String>()
        for (pair in cookiesString.split(";\\s*".toRegex())) {
            val index = pair.indexOf('=')
            if (index >= 0) {
                cookies[pair.substring(0, index)] = pair.substring(index + 1)
            }
        }
        return cookies
    }
}
